sklearn/preprocessing/data.py

```diff
@@ -1549,6 +1567,20 @@ def _transform(self, X):
             raise ValueError("X needs to contain only non-negative integers.")
         n_samples, n_features = X.shape
 
+        if (self.handle_unknown == 'error' and
+            self.n_values == 'auto'):
+            for i in range(n_features):
+                found_classes = set(np.unique(X[:, i]))
+                train_classes = set(self.unique_samples_[i])
+
+                if not found_classes.issubset(train_classes):
+                    total_set = found_classes.union(train_classes)
+                    new_classes = total_set.difference(train_classes)
+
+                    msg = ("unknown categorical feature(s) present"
+                                "during transform : %s" % str(new_classes))
+                    raise ValueError(msg)
+
```

Comment:
Now, we have decided to handle this way, we can remove the block of code from L1595 to L1603 that checks if a feature is greater than `self.n_values`